### PR TITLE
fetch: make fullyReadBody sync

### PR DESCRIPTION
**Pull request by Uzlopak**
closer to the spec and avoiding an async call.

### CODE REVIEW

File: lib/web/fetch/util.js [hunk 1]
```diff
@@ -1096,30 +1092,39 @@ function isomorphicEncode (input) {
...
-async function readAllBytes (reader) {
+async function readAllBytes (reader, successSteps, failureSteps) {
   const bytes = []
   let byteLength = 0
 
-  while (true) {
-    const { done, value: chunk } = await reader.read()
+  try {
+    do {
+      const { done, value: chunk } = await reader.read()
```
Comment (Uzlopak) [inline lib/web/fetch/util.js:1104]: can reader.read() throw?

Comment (KhafraDev) [reply lib/web/fetch/util.js:1104]: yes

Comment (Uzlopak) [reply lib/web/fetch/util.js:1104]: thx 

than putting it into a try catch block makes sense.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**